shared/utils/il2cpp-utils.cpp

```diff
@@ -94,8 +168,8 @@ namespace il2cpp_utils {
         
         if (!klass) return nullptr;
         // Check Cache
-        auto innerPair = std::pair<std::string, int>(methodName.data(), argsCount);
-        auto key = std::pair<Il2CppClass*, std::pair<std::string, int>>(klass, innerPair);
+        auto innerPair = std::pair<std::string, decltype(MethodInfo::parameters_count)>(methodName.data(), argsCount);
```

Comment:
Same deal here, why use decltype?

Reply:
We aim to update this library as rarely as possible, right?
So why not just use the types we're aiming for instead of our own?

More in-depth, though, the decltype (which is uint8_t) will actually save space in the hashmap compared to int.